src/Models/RememberModel.php

```diff
@@ -24,25 +25,36 @@ class RememberModel extends Model
      * Stores a remember-me token for the user.
      *
      * @param int|string $userId
-     *
-     * @return mixed
      */
-    public function rememberUser($userId, string $selector, string $hashedValidator, string $expires)
+    public function rememberUser($userId, string $selector, string $hashedValidator, string $expires): void
     {
         $expires = new DateTime($expires);
 
-        return $this->insert([
+        $return = $this->insert([
             'user_id'         => $userId,
             'selector'        => $selector,
             'hashedValidator' => $hashedValidator,
             'expires'         => $expires->format('Y-m-d H:i:s'),
         ]);
+
+        $this->checkQueryReturn($return);
+    }
+
+    private function checkQueryReturn(bool $return): void
+    {
+        if ($return === false) {
+            $error   = $this->db->error();
+            $message = 'Query error: ' . $error['code'] . ', '
+                . $error['message'] . ', query: ' . $this->db->getLastQuery();
+
+            throw new RuntimeException($message, $error['code']);
+        }
```

Comment:
The similarities here make me wonder if we should consider a central error handling library. Maybe for the next version? I know Lonnie had a very nice PR on the main repo expanding and centralizing exception handling that might be applicable.

Reply:
I think it is not good that the behavior of the database class changes depending on the value of `DBDebug`.
It should always throw an Exception on database errors.

Reply:
Agree! But as you know from recent conversations, I'm much more into throwing exceptions 🤗